applications/legend_formatter.py

```diff
@@ -0,0 +1,111 @@
+import json
+import lxml.html
+
+def filter_legend_data(legend_desc_shape, noise):
+	filtered_legend = [x for x in legend_desc_shape if x not in noise]
+	return filtered_legend
+
+def find_color(root, node_legend_length='none', edge_legend_length='none'):
+	color_shape = []
+	count = 0
+	for element in root.iter('font'):
+		count += 1
+		x = element.attrib['color'].replace('\"','')
+		color_shape.append(x.replace('\\',''))
+		if(count == node_legend_length + edge_legend_length):
+			return color_shape
+
+def convert_unicode_node_shape(node_shape):
+	ACCEPTED_NODE_SHAPES = {u'\u25a0':"rectangle", u'\u25b2':"triangle", u'\u25cf':"circle", u'\u2605': "star", u'\u25fe': "square"}
+	new_node_shape = []
+	for x in node_shape:
+		new_node_shape.append(ACCEPTED_NODE_SHAPES[x])
+	return new_node_shape
```

Comment:
I would suggest using List Comprehension for this case  :- 
`return [ACCEPTED_NODE_SHAPES[x] for x in node_shape]`